src/orchestrator/unary.rs

```diff
@@ -65,82 +65,85 @@ impl Orchestrator {
         let trace_id = task.trace_id;
         let headers = task.headers;
         info!(config = ?task.guardrails_config, "handling classification with generation task");
-        let task_handle = tokio::spawn(async move {
-            let input_text = task.inputs.clone();
-            let masks = task.guardrails_config.input_masks();
-            let input_detectors = task.guardrails_config.input_detectors();
-            // Do input detections
-            let input_detections = match input_detectors {
-                Some(detectors) if !detectors.is_empty() => {
-                    input_detection_task(
-                        &ctx,
-                        detectors,
-                        input_text.clone(),
-                        masks,
-                        headers.clone(),
-                    )
-                    .await?
-                }
-                _ => None,
-            };
-            debug!(?input_detections);
-            if let Some(mut input_detections) = input_detections {
-                // Detected HAP/PII
-                // Do tokenization to get input_token_count
-                let (input_token_count, _tokens) = tokenize(
-                    &ctx,
-                    task.model_id.clone(),
-                    task.inputs.clone(),
-                    headers.clone(),
-                )
-                .await?;
-                // Send result with input detections
-                input_detections.sort_by_key(|r| r.start);
-                Ok(ClassifiedGeneratedTextResult {
-                    input_token_count,
-                    token_classification_results: TextGenTokenClassificationResults {
-                        input: Some(input_detections),
-                        output: None,
-                    },
-                    warnings: Some(vec![InputWarning {
-                        id: Some(InputWarningReason::UnsuitableInput),
-                        message: Some(UNSUITABLE_INPUT_MESSAGE.to_string()),
-                    }]),
-                    ..Default::default()
-                })
-            } else {
-                // No HAP/PII detected
-                // Do text generation
-                let mut generation_results = generate(
-                    &ctx,
-                    task.model_id.clone(),
-                    task.inputs.clone(),
-                    task.text_gen_parameters.clone(),
-                    headers.clone(),
-                )
-                .await?;
-                debug!(?generation_results);
-                // Do output detections
-                let output_detectors = task.guardrails_config.output_detectors();
-                let output_detections = match output_detectors {
+        let task_handle = tokio::spawn(
```

Comment:
Just noticed - this file contains a lot of automatic reformatting because adding `.instrument(Span::current())` to async blocks (that need explicit instrumentation) causes `tokio::spawn(async move {` to break the `async move {` to a new line, incrementing indentation, iirc no functional changes besides the instrumentation